java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -118,7 +118,7 @@ public void onNext(final BarrageMessage barrageMessage) {
                 }
 
                 // override server-supplied data and regenerate local rowsets
-                barrageMessage.rowsAdded = RowSetFactory.flat(barrageMessage.length);
+                barrageMessage.rowsAdded = RowSetFactory.flat(barrageMessage.rowsIncluded.size());
```

Comment:
Don't we need to close the original `rowsAdded` and `rowsIncluded` if we're going to overwrite the pointers here?